VultisigApp/VultisigApp/Services/BalanceService.swift

```diff
@@ -20,78 +20,95 @@ class BalanceService {
     private let kuji = KujiraService.shared
     private let maya = MayachainService.shared
     private let dot = PolkadotService.shared
-
+    
+    private var cache: ThreadSafeDictionary<String, (data: (balance: String, priceRate: Double), timestamp: Date)> = ThreadSafeDictionary()
+    
+    private let CACHE_TIMEOUT_IN_SECONDS: Double = 120 // 2 minutes
+    
     func updateBalances(coins: [Coin]) async {
         await withTaskGroup(of: Void.self) { group in
             for coin in coins {
-                group.addTask { [unowned self]  in
+                group.addTask { [unowned self] in
                     if !Task.isCancelled {
                         await updateBalance(for: coin)
                     }
                 }
             }
         }
     }
-
+    
     @MainActor func updateBalance(for coin: Coin) async {
         do {
+            let cacheKey = "\(coin.ticker)-\(coin.contractAddress)-\(coin.chain.rawValue)-\(coin.address)"
```

Comment:
Please don't do this right now , there is no indication that getting balance or any api calls cause the hang

Reply:
Maybe just record what is the last time we do the balance check , if it is less than 1 minute , then we skip

Reply:
I will make it simple just setting the amount of coin rawBalance and PriceRate to match with the cached amount, just in case it missed somewhere.

Reply:
```
private var cache: ThreadSafeDictionary<String, (data: Bool, timestamp: Date)> = ThreadSafeDictionary()
let cacheKey = "\(coin.ticker)-\(coin.contractAddress)-\(coin.chain.rawValue)-\(coin.address)"
            if let cachedData = await Utils.getCachedData(cacheKey: cacheKey, cache: cache, timeInSeconds: CACHE_TIMEOUT_IN_SECONDS) {
                return
            }
```
I'm just using the cache to avoid going to APIs for few minutes and return whatever is already persisted in the coin.